### DISCUSSION TITLE
My link_to is not working.

### DISCUSSION THREAD

**Discussion by oporpino** [yippee-fun/phlex#609] [Help] [answered]
I'm trying to use link_to like this, but nothing is rendered on html response. 
`helpers.link_to("Show this client", @clients, class: "rounded-lg py-3 px-5 bg-gray-100 inline-block font-medium")`

Someone knows what could be?

**Comment by joeldrapper** [comment, marked as answer]
Hey, we already figured this out in the Naming Things Discord, but in case someone else finds this, I’ll answer here. 

The trick is to use the adapters rather than going through the `helpers` “proxy”, which is much lower-level. `helpers.link_to` calls the `link_to` helper directly and that helper returns HTML. This is because it's designed to be used with `<%= %>` output ERB tags.

In Phlex, you would expect a helper like `link_to` to output the HTML rather than return it. You don’t expect to have to pass it to `unsafe_raw` yourself.

To help with this, `phlex-rails` provides an adapter for every Rails helper. These adapters are provided as modules to include in your components as needed — though it probably makes sense to include your most frequently used adapters in your `ApplicationComponent`.

The `link_to` adapter, for example, is provided in the `Phlex::Rails::Helpers::LinkTo` module. Once you include that module, you can use `link_to` in your template.

```ruby
class MyComponent < ApplicationComponent
  include Phlex::Rails::Helpers::LinkTo

  def template
    link_to(root_path) { "Home" }
  end
end
```
